namerd/iface/interpreter-thrift/src/test/scala/io/buoyant/namerd/iface/ObserverCacheTest.scala

```diff
@@ -159,4 +161,34 @@ class ObserverCacheTest extends FunSuite {
     assert(activeSize(stats) == 0)
     assert(inactiveSize(stats) == 1)
   }
+
+  test("observer cache removes inactive observations after ttl time") {
+
+    val one = testObserver[String]()
+
+    val stats = new InMemoryStatsReceiver
+    val cache = new ObserverCache[String, String](1, 5, 1, stats, {
+      case "one" => one.obs
+    })
+
+    // insert "one" into the active cache
+    assert(cache.get("one").get eq one.obs)
+    assert(activeSize(stats) == 1)
+    assert(inactiveSize(stats) == 0)
+    // update deactivates "one"
+    one.event.notify(Return("foo"))
+
+    assert(activeSize(stats) == 0)
+    assert(inactiveSize(stats) == 1)
+    assert(!one.closed.sample)
+
+    implicit val patienceConfig: PatienceConfig = PatienceConfig(Span(3, Seconds), Span(1, Seconds))
```

Comment:
I think this interface can be used to make a faster and more robust test https://github.com/google/guava/wiki/CachesExplained#testing-timed-eviction

Reply:
Yes, but in our case, it requires exposing Ticker interface in the ObserverCache constructor. Do you want me to do that?